components/dashboard/src/start/VerifyModal.tsx

```diff
@@ -0,0 +1,220 @@
+                    },
+                });
+            }
+            return verified;
+        };
+
+        const reset = () => {
+            setState({
+                ...state,
+                sent: undefined,
+                message: undefined,
+                token: undefined,
+            });
+        };
+        return (
+            <Modal
+                onClose={() => {}}
+                closeable={false}
+                onEnter={verifyToken}
+                title="User Validation Required"
+                buttons={
+                    <div>
+                        <button className="ml-2" disabled={!isTokenFilled()} onClick={verifyToken}>
+                            Validate Account
+                        </button>
+                    </div>
+                }
+                visible={true}
+            >
+                <Alert type="warning" className="mt-2">
+                    To use Gitpod you'll need to validate your account with your phone number. This is required to
+                    discourage and reduce abuse on Gitpod infrastructure.
+                </Alert>
+                <div className="pt-4">
+                    <button className="gp-link" onClick={reset}>
+                        &larr; Use a different phone number
+                    </button>
+                </div>
+                <div className="pt-4">
+                    Enter the verification code we sent to {state.phoneNumber}.<br />
+                    Having trouble?{" "}
+                    <a className="gp-link" href="https://www.gitpod.io/contact/support">
+                        Contact support
+                    </a>
+                </div>
+                {state.message ? (
+                    <Alert type={state.message.type} className="mt-4 py-3">
+                        {state.message.text}
+                    </Alert>
+                ) : (
+                    <></>
+                )}
+                <div className="mt-4">
+                    <h4>Verification Code</h4>
+                    <input
+                        autoFocus={true}
+                        className="w-full"
+                        type="text"
+                        placeholder="Enter code sent via SMS"
+                        onChange={(v) => {
+                            setState({
+                                ...state,
+                                token: v.currentTarget.value,
+                            });
+                        }}
+                    />
+                </div>
+            </Modal>
+        );
+    } else {
+        const continueStartWorkspace = () => {
+            window.location.reload();
+            return true;
+        };
+        return (
+            <Modal
+                onClose={continueStartWorkspace}
+                closeable={false}
+                onEnter={continueStartWorkspace}
+                title="User Validation Successful"
```

Comment:
**suggestion**: Given it's not ideal to update the modal title based on the user action and taking into account the [relevant discussion](https://gitpod.slack.com/archives/C01KPEPQYE6/p1657728367916349?thread_ts=1657727187.914319&cid=C01KPEPQYE6) (internal), what do you think of moving the alert component on the loading screen directly where we later on show the warnings about the latest editor release and skip asking users to acknowledge the account validation?

| Latest Release on Workspace Start | Validation on Workspace Start | 
|-|-|
| <img width="1440" alt="Screenshot 2022-08-24 at 2 52 07 PM (2)" src="https://user-images.githubusercontent.com/120486/186412659-e3006b11-69b4-4a36-8d41-5f1c38e9e9ad.png"> | <img width="1440" alt="Screenshot 2022-08-24 at 2 51 18 PM (2)" src="https://user-images.githubusercontent.com/120486/186412663-3d27cf54-a06d-4181-8c93-e77baf801996.png"> |

Reply:
I think an explicit, "you are verified" message with a 'continue' button is clearer. Especially because we are reloading the page.

Reply:
**thought**: It is certainly clearer and gives users the time to acknowledge this.

**question**: What do you think of keeping modal, and updating the modal title to be the same on all cases, and rely on modal body (paragraph and alerts) to inform the user about validation errors or successful validation?